src/slurm_plugin/slurm_resources.py

```diff
@@ -397,8 +425,13 @@ def __init__(self, name, nodeaddr, nodehostname, state, partitions=None, reason=
 
     def is_state_healthy(self, terminate_drain_nodes, terminate_down_nodes, log_warn_if_unhealthy=True):
         """Check if a slurm node's scheduler state is considered healthy."""
-        # Check to see if node is in DRAINED, ignoring any node currently being replaced
-        if self.is_drained() and terminate_drain_nodes:
+        # Check if node is rebooting: if so, the node is healthy
+        if self.is_node_rebooting():
+            return True
+        # Check to see if node is in DRAINED, ignoring any node currently being replaced or in POWER_DOWN
+        if self.is_drained() and not self.is_power_down() and terminate_drain_nodes:
+            if self.is_power_down():
+                return True
```

Comment:
Why are we checking 
```
if self.is_power_down():
  return True
```
here?

Reply:
Thank you, I missed it in the refactoring.

Reply:
Addressed in the last commit.